src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,217 @@
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update "
+                    + car + " in carsDB.", e);
+        }
+        deleteDriversFromCarsById(car.getId());
+        insertDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                  PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT c.id AS car_id, model, "
+                + "name, country, m.id AS manufacturer_id "
+                + "FROM cars c "
+                + "JOIN cars_drivers cd "
+                + "ON c.id = cd.car_id "
+                + "JOIN manufacturers m "
+                + "ON c.manufacturer_id = m.id "
+                + "WHERE c.is_deleted = FALSE AND cd.driver_id = ?;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                  PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Car car = getCar(resultSet);
+                cars.add(car);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all cars by driver with id "
+                    + driverId, e);
+        }
+        for (Car car: cars) {
+            insertDrivers(car);
+        }
+        return cars;
+    }
+
+    private void insertDrivers(Car car) {
+        String insertRequest = "INSERT INTO cars_drivers (driver_id, car_id)"
+                + "VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                  PreparedStatement statement
+                          = connection.prepareStatement(insertRequest)) {
+            statement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                statement.setLong(2, driver.getId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't insert driver to data base. Car: " + car, e);
+        }
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        Manufacturer manufacturer = getManufacturer(resultSet);
+        Car car = new Car(resultSet.getString("model"), manufacturer, Collections.emptyList());
+        car.setId(resultSet.getLong("car_id"));
```

Comment:
it's better to use Object here car.setId(resultSet.getObject("car_id", Long.class))